### PR TITLE
Fix compiler warnings

### PR DESCRIPTION
**Pull request by smessmer**
Fix various compiler warnings

### CODE REVIEW

**Comment by bwasti** [comment]
I think this is broken for the qconv impl

**Comment by smessmer** [comment]
@bwasti It should be fixed now. Can you take another look?

File: torch/csrc/jit/custom_operator.h [hunk 1]
```diff
@@ -26,7 +26,10 @@ struct TORCH_API RegisterOperators {
   template <typename Implementation>
   C10_DEPRECATED_MESSAGE("torch::jit::RegisterOperators is deprecated. Please use torch::RegisterOperators instead.")
   RegisterOperators(const std::string& name, Implementation&& implementation) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
Comment (bwasti) [inline torch/csrc/jit/custom_operator.h]: do you think it's possible to avoid these?

**Review by bwasti — COMMENTED** [review]
Mostly looks good, thanks for doing this!

small comment on the pragma usage I'd like to resolve before accepting

File: torch/csrc/jit/custom_operator.h (hunk 1 above)
Comment (smessmer) [reply torch/csrc/jit/custom_operator.h]: Unfortunately, there doesn't seem to be a concept of "deprecated methods are allowed to call other deprecated methods", so whenever you call a deprecated method and don't want a warning, you have to add this - even if you're calling it from a method that's already deprecated.

This is ugly, I agree, but I don't have a better solution.

Comment (bwasti) [reply torch/csrc/jit/custom_operator.h]: the better approach is to remove all the deprecated invocations, right?  why are we even calling this method?

Comment (smessmer) [reply torch/csrc/jit/custom_operator.h]: Discussed offline and found a workaround. Thanks.

**Review by bwasti — APPROVED** [review]

**Pull request closed without merging**